### ISSUE TITLE
Cut & paste pastes block node that wasn't cut

### ISSUE THREAD

**Issue by DanweDE** [ianstormtaylor/slate#2427]
#### Do you want to request a _feature_ or report a _bug_?

bug

#### What's the current behavior?

1. Select a whole paragraph followed by a block node (image) by tripple-clicking it, [e.g. in this example](https://www.slatejs.org/#/images)
2. Cut the marked text (via cmd+x or context menu)
![cut-context](https://user-images.githubusercontent.com/101926/48566111-53312d80-e8fa-11e8-9bd4-e5956cb99520.gif)
3. Paste what you have previously cut in an empty editor:
![paste-context](https://user-images.githubusercontent.com/101926/48566452-1e71a600-e8fb-11e8-9501-003524ad0844.gif)

You will see the image was pasted even though it was not previously cut.

#### What's the expected behavior?

 - The block node (image) after the paragraph should not have a "selected" state after step 1.
 - When pasting in step 3, the image should not be pasted.

OR (less preferred):
 - Cutting should also cut the selected block node (image) after step 2

**Comment by ericedem** [comment]
I spent a while looking at this today, and I think I tracked some oddness down to this block of code:

https://github.com/ianstormtaylor/slate/blob/master/packages/slate-react/src/components/content.js#L223

On triple click in the exact example above, printing the `current` vs `range` looks like this:

![image](https://user-images.githubusercontent.com/1060669/48875029-4b1a3780-edab-11e8-80e7-dc75e92a52aa.png)

I couldn't find what is setting the end container of `current` to be inside the div after quite a while of tracing through, but I suspect that is part of the problem. Hopefully this gives someone else a better idea what's going on 😄

**Comment by isubasti** [comment]
This is more like triple click hanging selection problem, try it out on different browsers(I tried safari, chrome and firefox on osX) and they behave differently. firefox is the only one that works like we expected. The worst is actually safari, the selection even goes beyond the editor but when you cut/paste it have similar result to chrome.

**Comment by ianstormtaylor** [comment]
See https://github.com/ianstormtaylor/slate/issues/2746

**Comment by ianstormtaylor** [comment]
I believe that this may be fixed by https://github.com/ianstormtaylor/slate/pull/3093, which has changed a *lot* of the logic in Slate and `slate-react` especially. I'm going to close this out, but as always, feel free to open a new issue if it persists for you. Thanks for understanding.
